### PR TITLE
Create token by client certificate

### PR DESCRIPTION
**Pull request by molotkov-and**
* Improvement
Registration node request should be authorized by certificate and should not use token

### CODE REVIEW

File: ydb/core/grpc_services/auth_processor/dynamic_node_auth_processor.cpp [hunk 1]
```diff
@@ -63,6 +63,38 @@ TVector<std::pair<TString, TString>> X509CertificateReader::ReadSubjectTerms(con
...
+        if (!entry) {
+            continue;
+        }
+        const ASN1_STRING* data = X509_NAME_ENTRY_get_data(entry);
+        if (!data) {
+            continue;
+        }
+        int dataLen = (data->length > (bufLen - 1)) ? (bufLen - 1) : data->length;
+        memcpy(buf, data->data, dataLen);
+        buf[dataLen] = '\0';
```
Comment (adameat) [inline ydb/core/grpc_services/auth_processor/dynamic_node_auth_processor.cpp:85]: why do you need this intermediate buf ?

File: ydb/core/security/ticket_parser_impl.h [hunk 2]
```diff
@@ -644,6 +656,36 @@ class TTicketParserImpl : public TActorBootstrapped<TDerived> {
...
+        if (!x509cert) {
+            SetError(key, record, { .Message = error, .Retryable = false });
+            return false;
+        }
+        TStringBuilder dn;
+        for (const auto& [attribute, value] : X509CertificateReader::ReadAllSubjectTerms(x509cert)) {
+            dn << attribute << "=" << value << ",";
+        }
+        if (dn.empty()) {
+            SetError(key, record, { .Message = error, .Retryable = false });
```
Comment (adameat) [inline ydb/core/security/ticket_parser_impl.h:676]: it's better to have different error messages in different cases

File: ydb/core/security/ticket_parser_impl.h [hunk 3]
```diff
@@ -1497,14 +1541,14 @@ class TTicketParserImpl : public TActorBootstrapped<TDerived> {
...
         }
-
         if (tokenType == "Bearer" || tokenType == "IAM") {
             if (AccessServiceEnabled()) {
                 return TDerived::ETokenType::AccessService;
             } else {
                 return TDerived::ETokenType::Unsupported;
             }
-        } else if (tokenType == "ApiKey") {
+        }
```
Comment (adameat) [inline ydb/core/security/ticket_parser_impl.h:1550]: mmmm? why?

File: ydb/core/grpc_services/auth_processor/dynamic_node_auth_processor.cpp (hunk 1 above)
Comment (molotkov-and) [reply ydb/core/grpc_services/auth_processor/dynamic_node_auth_processor.cpp]: data consists of unsigned char and TString does not have constructor for unsigned char. Remove the buf, use reinterpret_cast<char*> for data

File: ydb/core/security/ticket_parser_impl.h (hunk 2 above)
Comment (molotkov-and) [reply ydb/core/security/ticket_parser_impl.h]: done

**Pull request closed without merging**